File: apps/web/src/entities/sprint/api/schemas.ts
import { z } from 'zod'

export const SprintCreateSchema = z.object({
  teamUuid: z.string(),
  label: z.string(),
  startsOn: z.string(),
  endsOn: z.string(),
  sharedGoal: z.string().nullable().optional(),
  periodLabel: z.string().nullable().optional(),
})
export type SprintCreate = z.infer<typeof SprintCreateSchema>

export const SprintReadSchema = z.object({
  uuid: z.string(),
  teamUuid: z.string(),
  label: z.string(),
  startsOn: z.string(),
  endsOn: z.string(),
  sharedGoal: z.string().nullable(),
  periodLabel: z.string().nullable(),
  createdAt: z.string(),
  updatedAt: z.string(),
})
export type SprintRead = z.infer<typeof SprintReadSchema>

export const SprintUpdateSchema = z.object({
  label: z.string().optional(),
  startsOn: z.string().optional(),
  endsOn: z.string().optional(),
  sharedGoal: z.string().nullable().optional(),
  periodLabel: z.string().nullable().optional(),
})
export type SprintUpdate = z.infer<typeof SprintUpdateSchema>

File: server/src/app/sprint/schemas.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

# -- Sprint --


class SprintCreate(BaseModel):
    team_uuid: str
    label: str
    starts_on: datetime
    ends_on: datetime
    shared_goal: str | None = None
    period_label: str | None = None


class SprintRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    uuid: str
    team_uuid: str
    label: str
    starts_on: datetime
    ends_on: datetime
    shared_goal: str | None
    period_label: str | None
    created_at: datetime
    updated_at: datetime


class SprintUpdate(BaseModel):
    label: str | None = None
    starts_on: datetime | None = None
    ends_on: datetime | None = None
    shared_goal: str | None = None
    period_label: str | None = None
